api/v1beta1/types.go

```diff
@@ -135,6 +135,8 @@ func (v *VnetSpec) IsManaged(clusterName string) bool {
 }
 
 // Subnets is a slice of Subnet.
+// +listType=map
+// +listMapKey=role
```

Comment:
why are we using role as map key? role isn't unique but name is so name might be a more appropriate map key if map key needs to be unique

Reply:
`name` seemed to work here (controller-gen didn't complain, at least), updated

Reply:
The problem with name is that it doesn't exist in the AzureClusterTemplate struct

Reply:
@sbueringer are you thinking of this type?

```
// SubnetTemplatesSpec specifies a list of subnet templates.
// +listType=map
// +listMapKey=name
type SubnetTemplatesSpec []SubnetTemplateSpec
```

Indeed the `SubnetTemplateSpec` type doesn't have a `Name` property, I wonder why code generation isn't complaining...

Reply:
Probably the code generation isn't verifying that. But looks like the apiserver is:

> unable to install CRDs onto control plane: unable to create CRD instances: unable to create CRD "azureclustertemplates.infrastructure.cluster.x-k8s.io": CustomResourceDefinition.apiextensions.k8s.io "azureclustertemplates.infrastructure.cluster.x-k8s.io" is invalid: spec.validation.openAPIV3Schema.properties[spec].properties[template].properties[spec].properties[networkSpec].properties[subnets].x-kubernetes-list-map-keys: Invalid value: []string{"name"}: entries must all be names of item properties

Reply:
I think we can resolve this conversation in favor of the other one